packages/dev/core/src/FlowGraph/flowGraphExecutionBlock.ts

```diff
@@ -0,0 +1,41 @@
+import type { FlowGraph } from "./flowGraph";
+import { FlowGraphBlock } from "./flowGraphBlock";
+import { FlowGraphConnectionPointDirection, FlowGraphSignalConnectionPoint } from "./flowGraphConnectionPoint";
+
+/**
+ * @experimental
+ * A block that executes some action. Always has an input signal.
+ * Can have one or more output signals.
+ */
+export abstract class FlowGraphExecutionBlock extends FlowGraphBlock {
+    /**
+     * The input signal of the block.
+     */
+    public readonly onStart: FlowGraphSignalConnectionPoint;
+
+    private readonly _signalInputs: FlowGraphSignalConnectionPoint[] = [];
+    private readonly _signalOutputs: FlowGraphSignalConnectionPoint[] = [];
+
+    constructor(graph: FlowGraph) {
+        super(graph);
+
+        this.onStart = this._registerSignalInput("onStart");
+    }
+    /**
+     * @internal
+     * Executes the flow graph execution block.
+     */
+    public abstract _execute(): void;

```

Comment:
A question about async execution - what happens when execute is async? is async expected to return immediately and let promises (for example) continue with the execution?